packages/splits-sdk/src/types.ts

```diff
@@ -117,6 +118,174 @@ export type GetSplitBalanceConfig = {
   token?: string
 }
 
+// Warehouse
+export type WarehouseTransferConfig = {
+  receiver: Address
+  token: Address
+  amount: bigint
+} & TransactionOverridesDict
+
+export type WarehouseTransferFromConfig = {
+  sender: Address
+  receiver: Address
+  token: Address
+  amount: bigint
+} & TransactionOverridesDict
+
+export type WarehouseApproveConfig = {
+  spender: Address
+  token: Address
+  amount: bigint
+} & TransactionOverridesDict
+
+export type WarehouseSetOperatorConfig = {
+  operator: Address
+  approved: boolean
+} & TransactionOverridesDict
+
+export type WarehouseInvalidateNonceConfig = {
+  nonce: bigint
+} & TransactionOverridesDict
+
+export type WarehouseTemporaryApproveAndCallConfig = {
+  spender: Address
+  operator: boolean
+  token: Address
+  amount: bigint
+  target: Address
+  data: Hex
+} & TransactionOverridesDict
+
+export type WarehouseTemporaryApproveAndCallBySigConfig = {
+  owner: Address
+  spender: Address
+  operator: boolean
+  token: Address
+  amount: bigint
+  target: Address
+  data: Hex
+  nonce: bigint
+  deadline: number
+  signature: Hex
+} & TransactionOverridesDict
+
+export type WarehouseApproveBySig = {
+  spender: Address
+  operator: boolean
+  token: Address
+  amount: bigint
+  nonce: bigint
+  deadline: number
+}
+
+export type WarehouseTemporaryApproveAndCallBySig = {
+  spender: Address
+  operator: boolean
+  token: Address
+  amount: bigint
+  target: Address
+  data: Hex
+  nonce: bigint
+  deadline: number
+}
+
+export type WarehouseApproveBySigConfig = {
+  owner: Address
+  spender: Address
+  operator: boolean
+  token: Address
+  amount: bigint
+  nonce: bigint
+  deadline: number
+  signature: Hex
+} & TransactionOverridesDict
+
+export type WarehouseDepositConfig = {
+  receiver: Address
+  token: Address
+  amount: bigint
+} & TransactionOverridesDict
+
+export type WarehouseBatchDepositConfig = {
+  receivers: Address[]
+  token: Address
+  amounts: bigint[]
+} & TransactionOverridesDict
+
+export type WarehouseWithdrawConfig = {
+  owner: Address
+  token: Address
+} & TransactionOverridesDict
+
+export type WarehouseBatchWithdrawConfig = {
+  owner: Address
+  tokens: Address[]
+  amounts: bigint[]
+  withdrawer: Address
+} & TransactionOverridesDict
+
+export type WarehouseBatchTransferConfig = {
+  receivers: Address[]
+  token: Address
+  amounts: bigint[]
+} & TransactionOverridesDict
+
+export type WarehouseSetWithdrawConfig = {
+  incentive: number
+  paused: boolean
+} & TransactionOverridesDict
+
+export enum SplitV2Type {
+  Push,
+  Pull,
+}
+
+// Split V2
+export type CreateSplitV2Config = {
+  recipients: Address[]
+  allocations: bigint[]
```

Comment:
In v1 we had people pass these in as percents. I guess that doesn't work anymore though? Cause the total allocation isn't always 1,000,000 anymore...

Just trying to think if there's a way to make this simple for users. I guess it's pretty simple though? Like users could pass in 50/30/20 instead of 500000/300000/200000 and it's the same either way, right?

I guess the biggest difference now is instead of decimals (i.e. 50.5) you always need to pass whole numbers (since they are bigints).

Reply:
yea this is worth a discussion post standup (or in person). my personal opinion is the sdk should be flexible but opinionated (lean more towards opinionated than flexible). i think it should make the common use cases as easy and simple as possible (and a more complex use case can either be not supported or a bit more complex/hard-to-use if it is supported).

if someone wants to interact directly with the contracts they are always free to do so, the contracts themselves are publicly available and well documented. The sdk is really to make life as easy as possible for developers to integrate in their apps.

Some of that complexity was already simplified by the v2 contracts (i.e. don't need to sort recipients anymore).

Reply:
Have some ideas, but yea let's discuss live tomorrow or thursday